### PR TITLE
feat: upgrade to React 19/Vite 8 and harden Tiptap editor lifecycle handling

### PR DESCRIPTION
**Pull request by ocruze**
## Summary

This PR modernizes the library and examples stack, improves runtime safety around Tiptap editor lifecycle edge cases, and aligns docs/tests with the new behavior.

### Tooling, compatibility, and CI
- Upgraded ecosystem versions across package/example workspaces (React 19 support, Vite 8, Tiptap v3.25 line, DSFR updates).
- Updated CI matrix to run on Node `20.19`, `22.12`, and `24`.
- Added CI steps for:
  - workspace type-check (`check-types`)
  - export validation (`check-exports`) for package entry points.
- Updated deploy workflow Node version to `20.19`.

### Package/export and dependency updates
- Updated `packages/react-dsfr-tiptap/package.json`:
  - modernized conditional exports structure (`default` + `types` mapping),
  - added/updated dependency and peer ranges (including React 18/19 compatibility),
  - introduced `check-exports:js` + `check-exports:css` and combined `check-exports`.

### Runtime hardening (destroyed editor guards)
Added defensive checks to avoid operations on destroyed/unmounted editors:
- `Provider.tsx`
- `ColorInput.tsx`
- `createControls.tsx`
- dialog components:
  - `LinkDialog.tsx`
  - `ImageDialog.tsx`
  - `YoutubeDialog.tsx`
- context typing cleanup in `contexts/dialog.ts` for React 19 type compatibility.

### Extension behavior consistency
- Default editor extension constants now use `StarterKit.configure({ link: false, underline: false })` to avoid duplicate extension registration when dedicated link/underline extensions are loaded.

### Tests
Expanded regression coverage for:
- missing/duplicate extension warning behavior,
- markdown/rich text link/underline scenarios,
- destroyed-editor safety paths in controls.

### Docs/examples
- Updated examples and README to reflect:
  - compatibility matrix (Node/React),
  - extension loading expectations,
  - StarterKit configuration guidance for link/underline,
  - corrected docs path references.

## Breaking/behavioral notes

- **Behavioral change (documented):** Link/Underline rely on dedicated extension loading paths; `StarterKit` defaults are configured to disable link/underline to prevent duplicate-extension warnings.
- Consumers using custom extension sets should ensure `link`/`underline` are configured consistently with the documented examples.

## Validation

- Lint
- Tests
- Type-check
- Build
- Export checks
- Examples build

### CODE REVIEW

File: examples/tsconfig.app.json [hunk 1]
```diff
@@ -9,6 +9,12 @@
 
         /* Bundler mode */
         "moduleResolution": "bundler",
+        "paths": {
+            "react-dsfr-tiptap": ["../packages/react-dsfr-tiptap/src/index.ts"],
+            "react-dsfr-tiptap/dialog": ["../packages/react-dsfr-tiptap/src/dialog.ts"],
+            "react-dsfr-tiptap/markdown": ["../packages/react-dsfr-tiptap/src/markdown.ts"],
+            "react-dsfr-tiptap/index.css": ["../packages/react-dsfr-tiptap/src/index.css"]
+        },
```
[the remark this answers is not in the document]
Comment (ocruze) [reply examples/tsconfig.app.json:17]: @copilot we're on typescript 5.9, baseUrl is deprecated, check again
